cmd/podman/manifest_annotate.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	manifestAnnotateCommand     cliconfig.ManifestAnnotateValues
+	manifestAnnotateDescription = `adds or updates information about an entry in a manifest list or image index`
+	_manifestAnnotateCommand    = &cobra.Command{
+		Use:   "annotate [flags] [manifest] [tags]",
+		Short: "manifest annotate",
+		Long:  manifestAnnotateDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			manifestAnnotateCommand.InputArgs = args
+			manifestAnnotateCommand.GlobalFlags = MainGlobalOpts
+			manifestAnnotateCommand.Remote = remoteclient
+			return manifestAnnotateCmd(&manifestAnnotateCommand)
+		},
+		Example: `podman manifest annotate --annotation left=right mylist:v1.11 image:v1.11-amd64`,
+		Args:    cobra.MinimumNArgs(2),
+	}
+)
+
+func init() {
+	manifestAnnotateCommand.Command = _manifestAnnotateCommand
+	manifestAnnotateCommand.SetHelpTemplate(HelpTemplate())
+	manifestAnnotateCommand.SetUsageTemplate(UsageTemplate())
+	flags := manifestAnnotateCommand.Flags()
+	flags.StringVar(&manifestAnnotateCommand.Os, "os", "", "override the `OS` of the specified image")
+	flags.StringVar(&manifestAnnotateCommand.Arch, "arch", "", "override the `Architecture` of the specified image")
+	flags.StringVar(&manifestAnnotateCommand.Variant, "variant", "", "override the `Variant` of the specified image")
+	flags.StringVar(&manifestAnnotateCommand.OsVersion, "os-version", "", "override the os `version` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateCommand.Features, "features", nil, "override the `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateCommand.OsFeatures, "os-features", nil, "override the os `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateCommand.Annotations, "annotation", nil, "set an `annotation` for the specified image")
+}
+
+func manifestAnnotateCmd(c *cliconfig.ManifestAnnotateValues) error {
+	listImageSpec := ""
+	imageSpec := ""
+	args := c.InputArgs
+	switch len(args) {
+	case 0:
+		return errors.New("At least a list image must be specified")
+	case 1:
+		listImageSpec = args[0]
+		if listImageSpec == "" {
+			return errors.Errorf(`Invalid image name "%s"`, args[0])
+		}
+	case 2:
+		listImageSpec = args[0]
+		if listImageSpec == "" {
+			return errors.Errorf(`Invalid image name "%s"`, args[0])
+		}
+		imageSpec = args[1]
+		if imageSpec == "" {
+			return errors.Errorf(`Invalid image name "%s"`, args[1])
+		}
+	default:
+		return errors.New("At least two arguments are necessary: list and image to add to list")
+	}
+
+	runtime, err := libpodruntime.GetRuntime(getContext(), &c.PodmanCommand)
+	if err != nil {
+		return errors.Wrapf(err, "could not create runtime")
+	}
+	defer runtime.DeferredShutdown(false)
+
+	store := runtime.GetStore()
+	systemContext := runtime.SystemContext()
+
+	_, listImage, err := util.FindImage(store, "", systemContext, listImageSpec)
```

Comment:
we typically use something like runtime.LookupImage (or something close to that.)